cider/tests/utils/CiderBatchStringifier.cpp

```diff
@@ -92,6 +95,51 @@ std::string StructBatchStringifier::stringifyValueAt(CiderBatch* batch, int row_
   return row.getString();
 }
 
+uint8_t DecimalBatchStringifier::getScale(const ScalarBatch<__int128_t>* batch) {
+  auto type_str = std::string(batch->getArrowFormatString());
+  uint8_t scale = std::stoi(type_str.substr(type_str.find(',') + 1));
+  return scale;
+}
+
+uint8_t DecimalBatchStringifier::getWidth(const ScalarBatch<__int128_t>* batch) {
+  auto type_str = std::string(batch->getArrowFormatString());
+  auto start = type_str.find(':') + 1;
+  auto end = type_str.find(',');
+  uint8_t width = std::stoi(type_str.substr(start, end - start));
+  return width;
+}
+
+std::string DecimalBatchStringifier::stringifyValueAt(CiderBatch* batch, int row_index) {
+  auto scalar_batch = batch->as<ScalarBatch<__int128_t>>();
+  if (!scalar_batch) {
+    CIDER_THROW(CiderRuntimeException,
+                "ScalarBatch is nullptr, maybe check your casting?");
+  }
+
+  auto scale = getScale(scalar_batch);
+  auto width = getWidth(scalar_batch);
+  auto data_buffer = scalar_batch->getRawData();
+  auto valid_bitmap = scalar_batch->getNulls();
+
+  if (valid_bitmap && !CiderBitUtils::isBitSetAt(valid_bitmap, row_index)) {
+    return NULL_VALUE;
+  } else {
+    __int128_t value = data_buffer[row_index];
+
+    if (!scale) {
+      // integral type
+      return CiderInt128Utils::Int128ToString(value);
+    } else {
```

Comment:
why not call Decimal128ToString?

Reply:
Mainly for consistency with stringifying `double`s. `double`s are converted to strings with 16 significant digits, but the number of significant digits of Decimal128ToString is determined by `scale`. So we first convert a decimal to `double`, and then stringify it to ensure consistency with the results in Cider (which are stored as `double`s)